src/services/fileStorage/docs/index.js

```diff
@@ -1,200 +1,895 @@
+/* eslint max-len: 0 */
 module.exports = {
 	fileStorageService: {
-		description: 'A service to handle external file storages',
-		create: {
-			parameters: [
-				{
-					description: 'the id of the school for which the storage will be created',
-					required: true,
-					name: 'schoolId',
-					type: 'string',
-				},
-			],
-			summary: 'Creates a new storage environment for a given school',
-			notes: 'Returns meta data of the created storage',
-		},
-		find: {
-			parameters: [
-				{
-					description: 'the context of the file-storage',
-					required: true,
-					name: 'path',
-					type: 'string',
-				},
-			],
-			summary: 'Retrieve all files and immediate sub-directories for the given path',
-		},
-		remove: {
-			parameters: [
-				{
-					description: 'The path where the file can be found',
-					required: true,
-					name: 'path',
-					type: 'string',
-				},
-			],
-			summary: 'remove a file',
-		},
-		patch: {
-			parameters: [
-				{
-					description: 'The name of the file',
-					required: true,
-					name: 'fileName',
-					type: 'string',
-				},
-				{
-					description: 'The path where the file can be found, have to have a slash at the end',
-					required: true,
-					name: 'path',
-					type: 'string',
-				},
-				{
-					description: 'The path where the file should be moved to, have to have a slash at the end',
-					required: true,
-					name: 'destination',
-					type: 'string',
-				},
-				{
-					description: 'The id of the file in the proxy db',
-					required: true,
-					name: 'id',
-					type: 'string',
-				},
-			],
-			summary: 'moves a file to a new path',
+		description: '',
+		operations: {
+			create: {
+				parameters: [
+					{
+						description: 'Filename',
+						name: 'name',
+						type: 'String',
+						required: true,
+					},
+					{
+						description: 'ID of either an course or team. For user files leave out.',
+						type: 'String (ObjectID)',
+						name: 'owner',
+					},
+					{
+						description: 'ID of parent folder',
+						type: 'String (ObjectID)',
+						name: 'parent',
+					},
+					{
+						description: 'Mimetype of file',
+						type: 'String',
+						name: 'type',
+					},
+					{
+						description: 'Filesize in Byte',
+						name: 'size',
+						type: 'Number',
+						schema: {
+							type: 'integer',
+							format: 'int32',
+							minimum: 0,
+						},
+					},
+					{
+						description: 'Filename aka identifier in storage context',
+						name: 'storageFileName',
+						type: 'String',
+						required: true,
+						schema: {
+							type: 'string',
+						},
+					},
+					{
+						description: 'Thumbnail of file',
+						type: 'String',
+						name: 'thumbnail',
+					},
+					{
+						description: 'Determine if students can edit if file is a course file',
+						type: 'Boolean',
+						name: 'studentCanEdit',
+						schema: {
+							type: 'boolean',
+						},
+					},
+					{
+						description: 'Override default permissions on creation',
```

Comment:
how to override something on creation?

Reply:
Because the default permissions are provided by the team.